toplevel/toploop.ml

```diff
@@ -145,32 +145,122 @@ let _ =
   Sys.interactive := true;
   Topeval.init ()
 
+(* Split a PATH-style variable, Windows-style. Entries are separated by
+   semicolons. Sections of entries may be double-quoted (which allows
+   semicolons in filenames to be quoted). The double-quote characters are
+   stripped (i.e. [f"o"o = foo]) *)
+let split_path_win32 path =
+  (* Buffer for storing the current segment being scanned *)
+  let buf = Buffer.create 256 in
+  let get_contents () =
+    let s = Buffer.contents buf in
+    Buffer.clear buf;
+    s
+  in
+  let add_segment segment_begin i =
+    Buffer.add_substring buf path segment_begin (i - segment_begin)
+  in
+  let len = String.length path in
+  let[@tail_mod_cons] rec parse segment_begin terminator i =
+    if i >= len then
+      (* Done - return the last entry *)
+      [get_contents (add_segment segment_begin i)]
```

Comment:
Just to confirm: in the case of `PATH=foo;` we want the result of this function to be `["foo";""]`? What are the semantics of `""` in `PATH`, is it the current directory?

Reply:
For XDG, it's moot - the first thing we do is filter the result for absolute paths (and `Filename.is_relative "" = true`, naturally). POSIX explicitly notes that an initial or final colon (or two consecutive colons) cause the current directory to be included in PATH, yes. On Windows, the CRT skips them, but that's an implementation detail - the interpretation of `PATH` is always that the current directory is searched first.